### PR TITLE
Add Unit tests for taxonomies

### PR DESCRIPTION
**Pull request by joemcgill**
### Issue Number
Related to #81.
~Blocked by #94.~

### Description
This adds several unit tests methods related to taxonomies.

- Adds method `test_get_url_list_custom_taxonomy()` that tests that custom taxonomies are returned by `Core_Sitemaps_Taxonomies::get_url_list()`
- Adds method `test_get_url_list_custom_taxonomy_private()` that tests that private custom taxonomies are _not_ returned by `Core_Sitemaps_Taxonomies::get_url_list()`
- Fixes a bug which allowed `Core_Sitemaps_Taxonomies::get_url_list()` to return non-public data
- Adds method `test_get_url_list_taxonomies()` which ensures that default taxonomy sitemaps include expected URL lists.
- Adds method `test_get_sitemap_entries_custom_taxonomies()`, which assures public custom taxonomies are included in the sitemap index and that private custom taxonomies are not.

### Screenshots (before and after if applicable)
If your PR includes visual changes include before and after screenshots showing the change.

### Type of change
Please select the relevant options:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (change which improves an existing feature. E.g., performance improvement, docs update, etc.)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Acceptance criteria
- [x] My code follows WordPress coding standards.
- [x] I have performed a self-review of my own code.
- [ ] If the changes are visual, I have cross browser / device tested.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added test instructions that prove my fix is effective or that my feature works.

### CODE REVIEW

File: tests/phpunit/class-test-core-sitemaps.php [hunk 1]
```diff
@@ -321,6 +331,40 @@ public function test_get_sitemap_entries_custom_post_types() {
...
+		// Create a custom public and private taxonomies for this test.
+		register_taxonomy( 'public_taxonomy', 'post' );
+		register_taxonomy( 'private_taxonomy', 'post', array( 'public' => false ) );
+
+		// Create test terms in the custom taxonomy.
+		$public_term  = $this->factory->term->create( array( 'taxonomy'  => 'public_taxonomy' ) );
+		$private_term = $this->factory->term->create( array( 'taxonomy'  => 'private_taxonomy' ) );
+
+		// Create a test post applied to all test terms.
+		$this->factory->post->create_and_get(
```
Comment (swissspidy) [inline tests/phpunit/class-test-core-sitemaps.php]:
```suggestion
		self::factory()->post->create_and_get(
```

File: tests/phpunit/class-test-core-sitemaps.php [hunk 2]
```diff
@@ -321,6 +331,40 @@ public function test_get_sitemap_entries_custom_post_types() {
...
+	public function test_get_sitemap_entries_custom_taxonomies() {
+		wp_set_current_user( self::$editor_id );
+
+		// Create a custom public and private taxonomies for this test.
+		register_taxonomy( 'public_taxonomy', 'post' );
+		register_taxonomy( 'private_taxonomy', 'post', array( 'public' => false ) );
+
+		// Create test terms in the custom taxonomy.
+		$public_term  = $this->factory->term->create( array( 'taxonomy'  => 'public_taxonomy' ) );
+		$private_term = $this->factory->term->create( array( 'taxonomy'  => 'private_taxonomy' ) );
```
Comment (swissspidy) [inline tests/phpunit/class-test-core-sitemaps.php]:
```suggestion
		$private_term = self::factory()->term->create( array( 'taxonomy'  => 'private_taxonomy' ) );
```

File: tests/phpunit/class-test-core-sitemaps.php [hunk 3]
```diff
@@ -321,6 +331,40 @@ public function test_get_sitemap_entries_custom_post_types() {
...
+	 */
+	public function test_get_sitemap_entries_custom_taxonomies() {
+		wp_set_current_user( self::$editor_id );
+
+		// Create a custom public and private taxonomies for this test.
+		register_taxonomy( 'public_taxonomy', 'post' );
+		register_taxonomy( 'private_taxonomy', 'post', array( 'public' => false ) );
+
+		// Create test terms in the custom taxonomy.
+		$public_term  = $this->factory->term->create( array( 'taxonomy'  => 'public_taxonomy' ) );
```
Comment (swissspidy) [inline tests/phpunit/class-test-core-sitemaps.php]:
```suggestion
		$public_term  = self::factory()->term->create( array( 'taxonomy'  => 'public_taxonomy' ) );
```

File: tests/phpunit/class-test-core-sitemaps.php [hunk 4]
```diff
@@ -422,6 +466,117 @@ public function test_get_url_list_cpt_private() {
...
+	/**
+	 * Test getting a URL list for default taxonomies via
+	 * Core_Sitemaps_Taxonomies::get_url_list().
+	 */
+	public function test_get_url_list_taxonomies() {
+		// Add the default category to the list of categories we're testing.
+		$categories = array_merge( array( 1 ), self::$cats );
+
+		// Create a test post to calculate update times.
+		$post = $this->factory->post->create_and_get(
```
Comment (swissspidy) [inline tests/phpunit/class-test-core-sitemaps.php]:
```suggestion
		$post = self::factory()->post->create_and_get(
```

File: tests/phpunit/class-test-core-sitemaps.php [hunk 5]
```diff
@@ -422,6 +466,117 @@ public function test_get_url_list_cpt_private() {
...
+	 */
+	public function test_get_url_list_custom_taxonomy() {
+		wp_set_current_user( self::$editor_id );
+
+		// Create a custom taxonomy for this test.
+		$taxonomy = 'test_taxonomy';
+		register_taxonomy( $taxonomy, 'post' );
+
+		// Create test terms in the custom taxonomy.
+		$terms = $this->factory->term->create_many( 10, array( 'taxonomy'  => $taxonomy ) );
```
Comment (swissspidy) [inline tests/phpunit/class-test-core-sitemaps.php]:
```suggestion
		$terms = self::factory()->term->create_many( 10, array( 'taxonomy'  => $taxonomy ) );
```

File: tests/phpunit/class-test-core-sitemaps.php [hunk 6]
```diff
@@ -422,6 +466,117 @@ public function test_get_url_list_cpt_private() {
...
+
+		// Create a test post applied to all test terms.
+		$post = $this->factory->post->create_and_get( array( 'tax_input' => array( $taxonomy => $terms ) ) );
```
Comment (swissspidy) [inline tests/phpunit/class-test-core-sitemaps.php]:
```suggestion
		$post = self::factory()->post->create_and_get( array( 'tax_input' => array( $taxonomy => $terms ) ) );
```

File: tests/phpunit/class-test-core-sitemaps.php [hunk 7]
```diff
@@ -422,6 +466,117 @@ public function test_get_url_list_cpt_private() {
...
+	 */
+	public function test_get_url_list_custom_taxonomy_private() {
+		wp_set_current_user( self::$editor_id );
+
+		// Create a custom taxonomy for this test.
+		$taxonomy = 'private_taxonomy';
+		register_taxonomy( $taxonomy, 'post', array( 'public' => false ) );
+
+		// Create test terms in the custom taxonomy.
+		$terms = $this->factory->term->create_many( 10, array( 'taxonomy'  => $taxonomy ) );
```
Comment (swissspidy) [inline tests/phpunit/class-test-core-sitemaps.php]:
```suggestion
		$terms = self::factory()->term->create_many( 10, array( 'taxonomy'  => $taxonomy ) );
```

File: tests/phpunit/class-test-core-sitemaps.php [hunk 8]
```diff
@@ -422,6 +466,117 @@ public function test_get_url_list_cpt_private() {
...
+
+		// Create a test post applied to all test terms.
+		$this->factory->post->create( array( 'tax_input' => array( $taxonomy => $terms ) ) );
```
Comment (swissspidy) [inline tests/phpunit/class-test-core-sitemaps.php]:
```suggestion
		self::factory()->post->create( array( 'tax_input' => array( $taxonomy => $terms ) ) );
```

**Comment by joemcgill** [comment]
@swissspidy I've updated this one to point to `master` now that #94 is merged and moved clean up routines to before assertions. Let me know if there is anything else to do here.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**